### ISSUE TITLE
ElementsCollection performance using iterators

### ISSUE THREAD

**Issue by CaBocuk** [codeborne/selenide#641]
## The problem
Once I had to iterate through an Elements collection of 100+ elements I faced that it takes too much time (~10 seconds).

## Details
So I investigated the issue and found that when we do the iteration - Selenide uses **SelenideElementIterator** to iterate **BySelectorCollection**. And it calls **BySelectorCollection's** method `getActualElements()` on EVERY `hasNext()` and `next()` call!

So i tried an experiment: I've changed the code of BySelectorCollection::getActualElements to:
```
  public List<WebElement> getActualElements() {
    SearchContext searchContext = parent == null ? getWebDriver() : parent;
    List<WebElement> elements = WebElementSelector.instance.findElements(searchContext, selector);
    System.out.println("Running "+ ++counter + " iteration of getActualElements(). Size = " +elements.size());
    return elements;
  }
```
and tested it on:
```
public static void main(String...args){
      Configuration.browser = "chrome";
      open("https://yandex.by/video/");
      $$("span").forEach(item -> item.is(visible));
  }
```

As an output I got: `Running 365 iteration of getActualElements(). Size = 182`
So for iterating through collection of 182 elements it finds 365 * 182 = **66430 elements**

Holding experiments with elements amount I figured out the the amount found elements is **(2*N^2 + N)**, where N - amount of elements in collection.
So for iterating through collection of 500 elements Selenide needs to find 500,500 elements :)

Should it really behave like that?

## Tell us about your environment
* **Selenide Version: 4.8**
* **Chrome: 62.0.3202.94**
* **Browser Driver Version: automatically downloaded**

**Comment by asolntsev** [comment]
@CaBocuk Thank you for reporting the issue. We definitely need to fix it.

**Comment by CaBocuk** [comment]
Fixed it in #653 
If the changes are ok, we can close it I think 😃
